### DISCUSSION TITLE
How can I use the password grant type?

### DISCUSSION THREAD

**Discussion by Dan6erbond** [authelia/authelia#4769] [Q&A] [answered]
Hey everyone, I'm trying to use the password grant type with Authelia to automate fetching access tokens. But for some reason, even though I've enabled the grant type in Authelia's config for my OIDC client, and followed this [guide](https://developer.okta.com/blog/2018/06/29/what-is-the-oauth2-password-grant) from Okta to request an access token, I get the following response:

```json
{
  "error": "invalid_request",
  "error_description": "The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed. Make sure that the various parameters are correct, be aware of case sensitivity and trim your parameters. Make sure that the client you are using has exactly whitelisted the redirect_uri you specified."
}
```

The error message is very ambigious, so I'm not sure where to begin with debugging the request I'm making.

These are the parameters I have. I'm making a Form encoded `POST` to `/api/oidc/token`:

```
grant_type=password
client_id=id
client_secret=secret
username=username
password=password
```

Has anyone gotten Authelia's password grant type working using an OIDC client? I appreciate the help!

**Comment by james-d-elliott** [comment, marked as answer]
We do not support Resource Owner Password Credentials as per the following resources:

- [OAuth 2.0 Security Best Current Practice: Resource Owner Password Credentials Grant](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics#name-resource-owner-password-cre)
- [OAuth 2.0 Specification: Password Grant](https://oauth.net/2/grant-types/password/)
- [OAuth 2.1  Specification (see the major differences section)](https://oauth.net/2.1/)
